### PR TITLE
Enhanced PropertyList to support types that allow multiple values for the same key

### PR DESCRIPTION
**Pull request by czardoz**

### CODE REVIEW

File: lib/collection/property-list.js [hunk 1]
```diff
@@ -110,7 +114,19 @@ _.assign(PropertyList.prototype, /** @lends PropertyList.prototype */ {
         if ((index = item[this._postman_listIndexKey]) && (index = String(index))) {
             // desensitise case, if the property needs it to be
             this._postman_listIndexCaseInsensitive && (index = index.toLowerCase());
-            this.reference[index] = item;
+
+            // if multiple values are allowed, the reference may contain an array of items, mapped to an index.
+            if (this._postman_propertyAllowsMultipleValues && this.reference[index]) {
```
Comment (shamasis) [inline lib/collection/property-list.js]: `this.reference.hasOwnProperty(index)`

File: lib/collection/property-list.js [hunk 2]
```diff
@@ -188,6 +204,8 @@ _.assign(PropertyList.prototype, /** @lends PropertyList.prototype */ {
             if (predicate.apply(context, arguments)) {
                 if ((index = item[this._postman_listIndexKey]) && (index = String(index))) {
                     this._postman_listIndexCaseInsensitive && (index = index.toLowerCase());
+
+                    // todo - figure out how it should work when we have _postman_propertyAllowsMultipleValues
```
Comment (shamasis) [inline lib/collection/property-list.js]: remove todo. this seems okay.

File: lib/collection/property-list.js [hunk 3]
```diff
@@ -227,13 +245,19 @@ _.assign(PropertyList.prototype, /** @lends PropertyList.prototype */ {
...
+     * Get Item in this list by `ID` reference. If multiple values are allowed, the last value is returned.
      *
      * @param {String} id
      * @returns {PropertyList.Type}
      */
     one: function (id) {
-        return this.reference[this._postman_listIndexCaseInsensitive ? String(id).toLowerCase() : id];
+        var val = this.reference[this._postman_listIndexCaseInsensitive ? String(id).toLowerCase() : id];
+
+        if (this._postman_propertyAllowsMultipleValues && _.isArray(val)) {
```
Comment (shamasis) [inline lib/collection/property-list.js]: see if you can banish lodash for this!

File: test/unit/header.test.js [hunk 4]
```diff
@@ -128,6 +128,22 @@ describe('Header', function () {
                 value: 'value2'
             }]);
         });
+        it.skip('must load a plain header key:value object, with values being an array', function () {
```
Comment (kunagpal) [inline test/unit/header.test.js]: Please add a todo.

File: lib/collection/property-list.js [hunk 5]
```diff
@@ -237,13 +253,19 @@ _.assign(PropertyList.prototype, /** @lends PropertyList.prototype */ {
...
      *
      * @param {String} id
      * @returns {PropertyList.Type}
      */
     one: function (id) {
-        return this.reference[this._postman_listIndexCaseInsensitive ? String(id).toLowerCase() : id];
+        var val = this.reference[this._postman_listIndexCaseInsensitive ? String(id).toLowerCase() : id];
+
+        if (this._postman_propertyAllowsMultipleValues && Array.isArray(val)) {
+            return val[val.length - 1];
```
Comment (shamasis) [inline lib/collection/property-list.js]: `val.length ? val[val.length - 1] : undefined`

Comment (shamasis) [inline lib/collection/property-list.js]: `val.length ? val[val.length - 1] : undefined`

**Pull request merged**